examples/node/stats.c

```diff
@@ -0,0 +1,144 @@
+/* Copyright (c) 2019, Codership Oy. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+
+#include "stats.h"
+
+#include "log.h"
+
+#include <assert.h>
+#include <stdio.h>  // snprintf()
+#include <stdlib.h> // abort()
+#include <unistd.h> // usleep()
+
+enum
+{
+    _STATS_REPL_BYTE,
+    _STATS_REPL_WS,
+    _STATS_RECV_BYTE,
+    _STATS_RECV_WS,
+    _STATS_TOTAL_BYTE,
+    _STATS_TOTAL_WS,
+    _STATS_CERT_FAILS,
+    _STATS_FC_PAUSED,
+    _STATS_MAX
+};
+
+static const char* const _stats_str[_STATS_MAX] =
+{
+    " repl(B/s)",
+    " repl(W/s)",
+    " recv(B/s)",
+    " recv(W/s)",
+    "total(B/s)",
+    "total(W/s)",
+    " cert.fail",
+    " paused(%)"
+};
+
+/* maps local stats IDs to provider stat IDs - provider dependent */
```

Comment:
This kind of mapping provider status variables can give bogus results. It would be better to look up by status variable name from status array.